### ISSUE TITLE
New CLI build with version 1.18.1 fails eslint

### ISSUE THREAD

**Issue by Peter-Piper** [oclif/oclif#710]
Do you want to request a *feature* or report a *bug*?
-----------------------------------------------------
Bug

What is the current behavior?
-----------------------------
Following the guide on how to generate a simple multi command cli and then running `yarn test` fails the eslint phase due to unicorn error

Version: oclif/1.18.1 darwin-x64 node-v14.17.4 (see [yarn.lock.txt](https://github.com/oclif/oclif/files/7511001/yarn.lock.txt) for further details)

How to reproduce

1. npx oclif multi mynewcli
2. . select yarn
3. answer yes to typescript
4. answer yes to eslint
5. answer yes to mocha
6. pres enter to skip cie
7. enter folder and run yarn test

```
➜  mynewcli yarn test
yarn run v1.22.11
$ nyc --extension .ts mocha --forbid-only "test/**/*.test.ts"



  hello
    ✓ runs hello (272ms)
    ✓ runs hello --name jeff


  2 passing (280ms)

----------|----------|----------|----------|----------|-------------------|
File      |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
----------|----------|----------|----------|----------|-------------------|
All files |    91.67 |       75 |      100 |    91.67 |                   |
 hello.ts |    91.67 |       75 |      100 |    91.67 |                11 |
----------|----------|----------|----------|----------|-------------------|
$ eslint . --ext .ts --config .eslintrc
TypeError: Error while loading rule 'unicorn/filename-case': context.getPhysicalFilename is not a function
Occurred while linting /Users/peterp/mynewcli/src/commands/hello.ts
    at create (/Users/peterp/mynewcli/node_modules/eslint-plugin-unicorn/rules/filename-case.js:156:40)
    at Object.wrapped [as create] (/Users/peterp/mynewcli/node_modules/eslint-plugin-unicorn/rules/utils/rule.js:38:18)
    at createRuleListeners (/Users/peterp/mynewcli/node_modules/eslint/lib/linter.js:578:21)
    at /Users/peterp/mynewcli/node_modules/eslint/lib/linter.js:732:31
    at Array.forEach (<anonymous>)
    at runRules (/Users/peterp/mynewcli/node_modules/eslint/lib/linter.js:690:34)
    at Linter._verifyWithoutProcessors (/Users/peterp/mynewcli/node_modules/eslint/lib/linter.js:899:31)
    at /Users/peterp/mynewcli/node_modules/eslint/lib/linter.js:955:35
    at Array.map (<anonymous>)
    at Linter.verify (/Users/peterp/mynewcli/node_modules/eslint/lib/linter.js:954:42)
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

What is the expected behavior?
------------------------------
I expect a completely cleanly generated oclif project will work from the beginning.

I really want to use oclif but I can't use 2.0 with generate (I've already created another ticker before for this) and I can't use 1.18.1

I hope we can get some help on this soon, as it is kind of blocking if we can't even follow the examples and how-to's of the repo

**Comment by f4z4onZH27** [comment]
I’m experiencing the same issue. I have found its cause and a temporary solution.

First, the cause. It’s the version range for `eslint-config-oclif`. App generators (`single`, `multi`, and `plugin`) in current stable add it as a dev dependency with `^3.1` version range (see [src/generators/app.ts:499](https://github.com/oclif/oclif/blob/v1.18.1/src/generators/app.ts#L499)). The problem is there’s a huge difference between dependencies of `eslint-config-oclif` version 3.1.0 and 3.1.1. Specifically, 3.1.0 depends on `eslint-plugin-unicorn@^6.0.1` whereas 3.1.1 depends on `eslint-plugin-unicorn@^36.0.0`. The impact of this is massive. One of the results is what we’re experiencing. It’s the reason the team released `eslint-config-oclif` a new major version — 4.0.0 introduces no changes.

My temporary solution is to apply [eslint-patch.txt](https://github.com/oclif/oclif/files/7614549/eslint-patch.txt) (my patch) and run `yarn`:

```bash
git apply eslint-patch.txt # or patch -p1 < eslint-patch.txt
yarn
```

To explain this a little, here’s equivalent and more laborious process:

1. Fixate `eslint-config-oclif` to 3.1.0 in `package.json`.
2. Run `yarn`.
3. Combine the new record for `eslint-config-oclif@3.1.0` with the record for `eslint-config-oclif@^3.1.0` in `yarn.lock`. Use the definition from `eslint-config-oclif@3.1.0`.
4. Run `yarn` again.

I know, it’s a bit extreme, but it works.

**Comment by f4z4onZH27** [comment]
I’ve just noticed that the [latest _dev_ release in v1 series](https://www.npmjs.com/package/oclif/v/1.18.2-dev.5) (v1.18.2-dev.5) doesn’t suffer from this issue. And it still uses `eslint-config-oclif@^3.1`. I guess this whole thing is probably about something else… 😅🤔

**Comment by Peter-Piper** [comment]
Thanks @f4z4onZH27 for replying. I thought I'd been forever ghosted 😄 
I'v try this out asap and hope it get's resolved in future releases (or hope it won't be an issue to v2.x as well)
/Cheers

**Comment by Peter-Piper** [comment]
@f4z4onZH27 Confirmed! This has been solved in v1.18.2-dev.5.

I will consider this ticket closed since this is will be part of next patch
